app/src/lib/rules/case_frame/common.js

```diff
@@ -337,26 +346,72 @@ export function validate_case_frame(tokens, trigger_index) {
 
 	// Show errors for missing and unexpected arguments
 	if (case_frame.missing_arguments.length) {
-		const missing_messages = case_frame.missing_arguments.map(rule => rule.missing_message)
-		
 		// TODO add appropriate error tokens instead of putting all the messages on the verb
```

Comment:
is this still needed?

Reply:
Yes, I'm thinking of inserting error tokens to put the message on instead of having all the messages on the verb, similar to how we add 'by X' when a passive is missing an agent. (Adding this functionality would actually replace that passive agent check)

For example, for be-O, if the agent is missing, insert the token [the weather] before the verb with the appropriate error message so the user knows what's required.

Reply:
oh ok, gotcha!